internal/engine/wazevo/backend/isa/arm64/lower_instr.go

```diff
@@ -674,6 +674,33 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
 		}
 		m.insert(dup)
 
+	case ssa.OpcodeScalarToVector:
+		x, lane := instr.ArgWithLane()
+		rd := operandNR(m.compiler.VRegOf(instr.Return()))
+
+		var rn operand
+		var arr vecArrangement
+		switch lane {
+		case ssa.VecLaneI32x4:
+			arr = vecArrangementS
+			rn = m.getOperand_NR(m.compiler.ValueDefinition(x), extModeNone)
+		case ssa.VecLaneI64x2:
+			arr = vecArrangementD
+			rn = m.getOperand_NR(m.compiler.ValueDefinition(x), extModeNone)
+		default:
+			panic("unsupported lane " + lane.String())
+		}
+
+		// movToVec does not zero-out the other lanes,
+		// so we first zero-out explicitly.
+		mov0 := m.allocateInstr()
+		mov0.asLoadFpuConst64(rd.nr(), 0)
+		m.insert(mov0)
```

Comment:
does this make sense or is there an instruction that does zero + movToVec all at once?

Reply:
interesting. then why are we passing the tests with the old compiler which seems to use one load? https://github.com/tetratelabs/wazero/blob/main/internal/engine/compiler/impl_vec_arm64.go#L247-L262

Reply:
Yeah I noticed and I am confused. I will investigate

Reply:
it should be ok now